analyses/batch-effects/util/functions.R

```diff
@@ -0,0 +1,361 @@
+# function used to cut files
+shorten = function(df) {
+  return(df[1:100,])
+}
+
+# function used on kallisto data for summarizing transcripts at the gene level
+grouper = function(df){
+  df = df %>%
+    group_by(gene_id) %>%
+    summarise_at(colnames(df)[3:ncol(df)], mean, na.rm = TRUE)
+  return(df)
+}
+
+# transpose data through gather and spread functions, rename the index column
+trans = function(data, Kids_First_Biospecimen_ID, val, starting_col){
+  data = data %>%
+    gather(Kids_First_Biospecimen_ID, val, starting_col:(ncol(data))) %>%
+    spread("gene_id", val)
```

Comment:
It is not clear to me what the advantage of this is over `t()`. Does this transpose only happen when you have a bunch of metadata alongside the expression values?

Reply:
I'll relook into t() instead of using gathering and spread. I wrote this function a while ago. I might have had to do it this way because of the ID columns, but I'm not sure. I'll send you an update later this week on this.